backend/app/api/v1/endpoints/cpt/cptJobs.py

```diff
@@ -25,161 +26,191 @@
-    size: int = Query(None, description="Number of jobs to fetch"),
-    offset: int = Query(None, description="Offset Number to fetch jobs from"),
-    totalJobs: int = Query(None, description="Total number of jobs"),
+    pretty: bool = Query(False, description="Pretty output format"),
+    size: int = Query(None, description="Number of jobs"),
+    offset: int = Query(None, description="Offset"),
+    filter: str = Query(None, description="Query to filter jobs"),
+    totalJobs: int = Query(None, description="Total job count"),
 ):
-    if start_date is None:
-        start_date = datetime.utcnow().date()
-        start_date = start_date - timedelta(days=5)
-
-    if end_date is None:
-        end_date = datetime.utcnow().date()
+    start_date, end_date = get_default_dates(start_date, end_date)
 
     if start_date > end_date:
         return Response(
-            content=json.dumps(
-                {"error": "invalid date format, start_date must be less than end_date"}
-            ),
+            content=json.dumps({"error": "start_date must be before end_date"}),
             status_code=422,
         )
 
     offset, size = normalize_pagination(offset, size)
 
-    results_df = pd.DataFrame()
-    total_dict = {}
-    total = 0
-    with ProcessPoolExecutor(max_workers=cpu_count()) as executor:
-        futures = {
-            executor.submit(
-                fetch_product, product, start_date, end_date, size, offset
-            ): product
+    results = await asyncio.gather(
+        *[
+            fetch_data_limited(product, start_date, end_date, size, offset, filter)
             for product in products
-        }
-        for future in as_completed(futures):
-            product = futures[future]
-            try:
-                result = future.result()
-                total_dict[product] = result["total"]
-                results_df = pd.concat([results_df, result["data"]])
-            except Exception as e:
-                print(f"Error fetching data for product {product}: {e}")
-
-    jobsCount = totalJobs
-    # The total is determined by summing the counts of all products and is included in the response.
-    # However, during pagination, if the count of any product drops to zero,
-    # the total becomes lower than the actual value, which is undesirable.
-
-    # on first hit, totalJobs is 0
-    if totalJobs == 0:
-        for product in total_dict:
-            total += int(total_dict[product])
-        jobsCount = total
-    response = {
-        "startDate": start_date.__str__(),
-        "endDate": end_date.__str__(),
-        "results": results_df.to_dict("records"),
-        "total": jobsCount,
-        "offset": offset + size,
-    }
+        ],
+        return_exceptions=True,
+    )
 
-    if pretty:
-        json_str = json.dumps(response, indent=4)
-        return Response(content=json_str, media_type="application/json")
+    results = [res for res in results if isinstance(res, dict)]
```

Comment:
If anything in the `results` list isn't a `dict`, should we really be throwing it away silently, or should we at least be reporting what it was??